# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.h
# Commit: ea27cd66fefd
# Full Hash: ea27cd66fefd175c69de110191319544a252e820
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-18 09:51:31
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - Support conversation between InternalRequest and IPCInternalRequest. r=dom-worker-reviewers,jesup
#   
#   Depends on D138812
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138813
# ==============================================================================

diff -r 8c21c9468949 -r ea27cd66fefd dom/fetch/InternalRequest.h
--- a/dom/fetch/InternalRequest.h	Tue Jan 17 15:13:44 2023 +0000
+++ b/dom/fetch/InternalRequest.h	Tue Jan 17 15:13:44 2023 +0000
@@ -29,6 +29,7 @@
 namespace mozilla {
 
 namespace ipc {
+class PBackgroundChild;
 class PrincipalInfo;
 }  // namespace ipc
 
@@ -97,6 +98,9 @@
 
   explicit InternalRequest(const IPCInternalRequest& aIPCRequest);
 
+  void ToIPCInternalRequest(IPCInternalRequest* aIPCRequest,
+                            mozilla::ipc::PBackgroundChild* aManager);
+
   SafeRefPtr<InternalRequest> Clone();
 
   void GetMethod(nsCString& aMethod) const { aMethod.Assign(mMethod); }